### PR TITLE
Add argument `include_file_paths` in `pl_scan_csv`

### PR DESCRIPTION
**Pull request by collioud**
**Context**
The argument `include_file_paths` is missing from the `pl_scan_csv`  and `pl_read_csv` functions whereas it is available for other `pl_read_...` fonctions (for example, `pl_read_parquet`).
Close [#1235 ](https://github.com/pola-rs/r-polars/issues/1235)

**Goal of this PR**
Add the argument `include_file_paths` to the `pl_scan_csv`  and `pl_read_csv` functions.
Adjust the related documentation.

**Tests**
Not tested yet

Hope this will help the project!
Cheers

### CODE REVIEW

File: tests/testthat/test-csv-read.R [hunk 1]
```diff
@@ -197,3 +197,15 @@ test_that("cache url tempfile", {
   expect_false(is.null(cache_temp_file[[url]]))
   expect_equal(attempt_1, attempt_2)
 })
+
+test_that("scan_csv can include file path", {
+  skip_if_not_installed("withr")
+  temp_dir = withr::local_tempdir()
```
Comment (etiennebacher) [inline tests/testthat/test-csv-read.R:203]: This should be `local_tempfile` instead

File: tests/testthat/test-csv-read.R [hunk 2]
```diff
@@ -197,3 +197,15 @@ test_that("cache url tempfile", {
...
+  pl$DataFrame(mtcars)$write_csv(temp_dir)
+
+  expect_identical(
+    pl$scan_csv(temp_dir, include_file_paths = "file_paths")$collect()$unique("file_paths") |>
+      dim(),
+    c(1L, 12L)
+  )
+})
```
Comment (etiennebacher) [inline tests/testthat/test-csv-read.R:211]: Could you modify the example to read two csv files at once? That way we can ensure that it returns two file paths and not the same all the time

**Review by etiennebacher — CHANGES_REQUESTED** [review]
Thanks, this looks good to me, there are just a couple of things to fix in tests.

Also, can you add a bullet point in NEWS.md indicating the new feature and the PR number (and your github username if you want)?

File: tests/testthat/test-csv-read.R (hunk 1 above)
Comment (collioud) [reply tests/testthat/test-csv-read.R]: Done

File: tests/testthat/test-csv-read.R (hunk 2 above)
Comment (collioud) [reply tests/testthat/test-csv-read.R:212]: Also done

**Comment by collioud** [comment]
Made a commit with the asked modifications.
Have a good night!

**Comment by etiennebacher** [comment]
Almost done, you just need to redocument. You can use `task build-documents` for that

**Comment by collioud** [comment]
Done. 
Please note that I changed a bit the `tests/testthat/test-csv-read.R` file to be coherent with `test-parquet.R`.
When doing `task build-documents`, I also have modifications in some doc files (like` man/Series_to_r.Rd`, `man/Series_class.Rd`, etc.), but it looks like it is unrelated to this PR, as far as I know. I let the changes, but feel free to discard them...

**Comment by etiennebacher** [comment]
I cleaned up the PR, I think the extra changes in `.Rd` files were due to the fact you don't have the latest version of `roxygen2`.

That's a nice first contribution, thanks @collioud!

**Review by etiennebacher — APPROVED** [review]

**Pull request merged**